pkgs/applications/networking/p2p/soulseekqt/default.nix

```diff
@@ -14,46 +16,52 @@ mkDerivation rec {
   pname = "soulseekqt";
   version = "2018-1-30";
 
-  src = fetchurl {
-      urls = [
-        "https://www.dropbox.com/s/0vi87eef3ooh7iy/SoulseekQt-${version}.tgz"
-        "https://www.slsknet.org/SoulseekQt/Linux/SoulseekQt-${version}-64bit-appimage.tgz"
-      ];
-      sha256 = "0d1cayxr1a4j19bc5a3qp9pg22ggzmd55b6f5av3lc6lvwqqg4w6";
+  src = fetchzip {
+      url = "https://www.slsknet.org/SoulseekQt/Linux/SoulseekQt-${version}-64bit-appimage.tgz";
+      sha256 = "16ncnvv8h33f161mgy7qc0wjvvqahsbwvby65qhgfh9pbbgb4xgg";
     };
 
   dontBuild = true;
 
-  nativeBuildInputs = [ autoPatchelfHook pythonPackages.binwalk squashfsTools desktop-file-utils ];
+  nativeBuildInputs = [ imagemagick radare2 jq autoPatchelfHook squashfsTools
+    desktop-file-utils ];
   buildInputs = [ qtmultimedia stdenv.cc.cc ];
 
   # avoid usage of appimage's runner option --appimage-extract
-  unpackCmd = ''
-    export HOME=$(pwd) # workaround for binwalk
-    appimage=$(tar xvf $curSrc) && binwalk --quiet \
-       $appimage -D 'squashfs:squashfs:unsquashfs %e'
-    '';
+  postUnpack = ''
+    cd $sourceRoot
+
+    # multiarch offset one-liner using same method as AppImage
+    offset=$(r2 *.AppImage -nn -Nqc "pfj.elf_header @ 0" |\
+      jq 'map({(.name): .value}) | add | .shoff + (.shnum * .shentsize)')
+
+    unsquashfs -o $offset *.AppImage
+    sourceRoot=squashfs-root
```

Comment:
this is very terse, what does this do and where is it the same method as AppImage for what?

Reply:
to Calculate the size of an ELF file on disk based on the information in its header, it's same as
https://gist.github.com/probonopd/a490ba3401b5ef7b881d5e603fa20c93